src/stores/AccountController.ts

```diff
@@ -141,66 +132,70 @@ const refreshJwtToken = async (sandbox: boolean, auth: AuthData) => {
   }
 };
 
-export const afterLogin = async (sandbox: boolean, auth: AuthData) => {
-  const { accessModel } = useConfigStore.getState();
-  const decodedToken: JwtDetails = jwtDecode(auth.jwt);
-  const customerId = decodedToken.customerId;
-  const response = await accountService.getCustomer({ customerId }, sandbox, auth.jwt);
-
-  if (response.errors.length) throw new Error(response.errors[0]);
+export const getAccount = async (auth: AuthData) => {
+  await withAccountService(async ({ accountService, config, accessModel }) => {
+    const response = await accountService.getUser({ config, auth });
 
-  useAccountStore.setState({
-    auth: auth,
-    user: response.responseData,
-  });
+    useAccountStore.setState({
+      auth: auth,
+      user: response,
+    });
 
-  await Promise.allSettled([accessModel === 'SVOD' ? reloadActiveSubscription() : Promise.resolve(), getCustomerConsents(), getPublisherConsents()]);
+    await getAccountExtras(accessModel);
 
-  useAccountStore.setState({ loading: false });
+    useAccountStore.setState({ loading: false });
+  });
 };
 
 export const login = async (email: string, password: string) => {
-  await useConfig(async ({ cleengId, cleengSandbox }) => {
+  await withAccountService(async ({ accountService, config, accessModel }) => {
     useAccountStore.setState({ loading: true });
 
-    const response = await accountService.login({ email, password, publisherId: cleengId }, cleengSandbox);
+    const response = await accountService.login({ config, email, password });
 
-    if (response.errors.length > 0) throw new Error(response.errors[0]);
+    useAccountStore.setState({
+      auth: response.auth,
+      user: response.user,
+    });
 
-    await afterLogin(cleengSandbox, response.responseData);
+    await getAccountExtras(accessModel);
 
     await restoreFavorites();
     await restoreWatchHistory();
+    useAccountStore.setState({ loading: false });
   });
 };
 
 export const logout = async () => {
-  persist.removeItem(PERSIST_KEY_ACCOUNT);
-
-  // this invalidates all entitlements caches which makes the useEntitlement hook to verify the entitlements.
-  await queryClient.invalidateQueries('entitlements');
-
-  useAccountStore.setState({
-    auth: null,
-    user: null,
-    subscription: null,
-    transactions: null,
-    activePayment: null,
-    customerConsents: null,
-    publisherConsents: null,
-  });
+  await withAccountService(async ({ accountService }) => {
+    persist.removeItem(PERSIST_KEY_ACCOUNT);
```

Comment:
Does the inplayer sdk need this or is similar functionality wrapped into the sdk already?  If it doesn't need it, maybe it makes sense to put this into the cleeng service's logout method?

Reply:
We adapted the entire logic about authentication as the current flow. We use the jwapp.auth as well for the app. The SDK localStorage is the same as the `jwapp.auth` but has no effect on the ott app. It comes out of the box with the SDK so that the endpoints can be tokenised.